cirq-ft/cirq_ft/infra/gate_with_registers.py

```diff
@@ -54,11 +67,14 @@ def __repr__(self):
 
     @property
     def bitsize(self) -> int:
-        return sum(reg.bitsize for reg in self)
+        return sum(reg.total_bits() for reg in self)
 
     @classmethod
-    def build(cls, **registers: int) -> 'Registers':
-        return cls(Register(name=k, bitsize=v) for k, v in registers.items())
+    def build(cls, **registers: Union[int, Tuple[int, ...]]) -> 'Registers':
+        return cls(
+            Register(name=k, shape=v if isinstance(v, tuple) else (v,))
```

Comment:
this conversion is either redundant with the attrs converter or subtly (and confusingly) different. 

```python
Registers.build(x=[5,6])
```

would give you something unexpected